sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/StorageFileInputStream.java

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file;
+
+import com.azure.core.implementation.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.common.Constants;
+import com.azure.storage.common.StorageInputStream;
+import com.azure.storage.file.models.FileRange;
+import com.azure.storage.file.models.StorageException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Provides an input stream to read a given storage file resource.
+ */
+public class StorageFileInputStream extends StorageInputStream {
+    final ClientLogger logger = new ClientLogger(StorageFileInputStream.class);
+
+    private final FileAsyncClient fileAsyncClient;
+
+    /**
+     * Holds the length of the current buffer in bytes.
+     */
+    private int bufferSize;
+
+    /**
+     * Holds the last exception this stream encountered.
+     */
+    private IOException lastError;
+
+    /**
+     * A flag to determine if the stream is faulted, if so the last error will be thrown on next operation.
+     */
+    private volatile boolean streamFaulted;
+
+    /**
+     * Holds the absolute byte position of the start of the current buffer.
+     */
+    private long bufferStartOffset;
+
+    /**
+     * Initializes a new instance of the StorageFileInputStream class.
+     *
+     * @param fileAsyncClient A {@link FileClient} object which represents the blob that this stream is associated with.
+     * @throws StorageException An exception representing any error which occurred during the operation.
+     */
+    StorageFileInputStream(final FileAsyncClient fileAsyncClient)
+        throws StorageException {
+        this(fileAsyncClient, 0, null);
+    }
+
+    /**
+     * Initializes a new instance of the StorageFileInputStream class. Note that if {@code fileRangeOffset} is not
+     * {@code 0} or {@code fileRangeLength} is not {@code null}, there will be no content MD5 verification.
+     *
+     * @param fileAsyncClient A {@link FileAsyncClient} object which represents the blob
+     * that this stream is associated with.
+     * @param fileRangeOffset The offset of file range data to begin stream.
+     * @param fileRangeLength How much data the stream should return after fileRangeOffset.
+     * @throws StorageException An exception representing any error which occurred during the operation.
+     */
+    StorageFileInputStream(final FileAsyncClient fileAsyncClient, long fileRangeOffset, Long fileRangeLength)
```

Comment:
>FileAsyncClient [](start = 33, length = 15)

sync client

Reply:
Prefer no change for this PR before preview